src/Workspaces/Core/Portable/Execution/PinnedRemotableDataScope.cs

```diff
@@ -63,7 +64,12 @@ public RemotableData GetRemotableData(Checksum checksum, CancellationToken cance
 
         public void Dispose()
         {
-            _storages.UnregisterSnapshot(this);
+            if (!_disposed)
```

Comment:
I do not believe double-disposing is currently possible. It was possible prior to the refactoring in 16ab0cdfd3003fca086b581130ddea3a501ca29e if cancellation occurred during the call to `session.InitializeAsync()`, with these two locations calling dispose:

1. The session would call `this.Dispose()` via the cancellation token registration.
2. The cancellation would get propagated as an exception up to `RemoteHostClient.TryCreateServiceSessionAsync`, where the exception handler would explicitly call `snapshot.Dispose()`.

This code path no longer exists, and it doesn't appear that double-dispose is possible. However, best practices for implementing `IDisposable` still include the recommendation that `Dispose()` be safe to call more than once, so I left this support in place.